cmd/multiFaAcceleration/multiFaAcceleration.go

```diff
@@ -0,0 +1,504 @@
+// Command Group: "Statistics & Population Genetics"
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fasta"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/numbers"
+	"log"
+	"math"
+)
+
+//Settings contains all program arguments and options, and is passed throughout helper functions to aid readability.
+type Settings struct {
+	InFile                     string
+	ChromName                  string
+	VelOut                     string
+	AccelOut                   string
+	InitialVelOut              string
+	SearchSpaceBed             string
+	SearchSpaceProportion      float64
+	WindowSize                 int
+	UseSnpDistance             bool
+	Verbose                    bool
+	Epsilon                    float64
+	AllowNegative              bool
+	ZeroDistanceWeightConstant float64
+}
+
+//Once we have branch lengths for each valid window, we will need to normalize the values relative to each other. Thus, we store the branch lengths in this intermediate cache before writing to file.
+type BranchCache struct {
+	ChromStart int
+	ChromEnd   int
+	b1         float64
+	b3         float64
+}
+
+//A set of all observed pairwise distances between the four species. While these numbers must be integers, we store them as float64 to avoid casting in other functions as a way to improve readability.
+type Distances struct {
+	D01 float64
+	D02 float64
+	D03 float64
+	D12 float64
+	D13 float64
+	D23 float64
+}
+
+//The set of branch lengths corresponding to a particular distance matrix.
+type BranchLengths struct {
+	B1 float64
+	B2 float64
+	B3 float64
+	B4 float64
+	B5 float64
+}
+
+//SubTree is a tree with three leaves and three branches, joined at the single internal node.
+type SubTree struct {
+	Dab float64
+	Dac float64
+	Dbc float64
+	va  float64
+	vb  float64
+	vc  float64
+}
+
+func multiFaAcceleration(s Settings) {
+	records := fasta.Read(s.InFile)
+	var searchSpace []bed.Bed
+	var referenceLength, i, j, threshold int
+	var bitArray []byte
+
+	//a couple of safety checks
+	if len(records) != 4 {
+		log.Fatalf("multiFaAcceleration accepts a multiFa file with 4 records, found %v.", len(records))
+	}
+	if len(records[1].Seq) != len(records[0].Seq) || len(records[2].Seq) != len(records[0].Seq) || len(records[3].Seq) != len(records[0].Seq) {
+		log.Fatalf("Error. All records must be of the same sequence length.")
+	}
+
+	referenceLength = fasta.AlnPosToRefPos(records[0], len(records[0].Seq)-1)
+
+	if s.SearchSpaceBed != "" {
+		//if we want to limit our search to a particular set of regions, we populate the bitArray with 1s instead of 0s at positions overlapping the input bed entries.
+		searchSpace = bed.Read(s.SearchSpaceBed)
+		bitArray = make([]byte, referenceLength)
```

Comment:
I was thinking that calling it bitArray was to let the reader know it was not being used as an array of bytes, but rather that each bit would be used for a position.  However, I am now thinking we are only using the least-significant bit in the byte.  Maybe it would be more clear if this was a slice of bools?